components/camel-spring-boot/src/main/java/org/apache/camel/spring/boot/TypeConversionConfiguration.java

```diff
@@ -33,14 +33,18 @@
 @ConditionalOnProperty(value = "camel.springboot.typeConversion", matchIfMissing = true)
 public class TypeConversionConfiguration {
 
-    @Bean(initMethod = "", destroyMethod = "")
+    // We explicitly declare the destroyMethod to be "" as the Spring @Bean
+    // annotation defaults to AbstractBeanDefinition.INFER_METHOD otherwise
+    // and in that case ShutdownableService::shutdown/Service::close
+    // (BaseTypeConverterRegistry extends ServiceSupport) would be used for
+    // bean destruction. And we want Camel to handle the lifecycle.
+    @Bean(destroyMethod = "")
     // Camel handles the lifecycle of this bean
     TypeConverter typeConverter(CamelContext camelContext) {
         return camelContext.getTypeConverter();
     }
 
-    @Bean(initMethod = "", destroyMethod = "")
-    // Camel handles the lifecycle of this bean
+    @Bean
```

Comment:
I've removed `initMethod` and `destroyMethod` as `SpringTypeConverter` is a concrete class and does not provide `stop` or `shutdown` methods that might be auto-configured for bean destruction.